woocommerce/woocommerce-gateway-stripe
# Render ECE button behind feature flag

Fixes #3402 

## Changes proposed in this Pull Request:
- Updated the `"@stripe/react-stripe-js"` version to the latest. Otherwise ECE buttons are not loaded by Stripe.
- Introduced a new option `_wcstripe_feature_ece` as the feature flag.
- In the block checkout page, render the ECE component when the feature flag is enabled. If feature flag is disabled then render the PRB component.
- This PR only displays the ECE button, no functionality has been added yet.
- ECE button is rendered on block checkout only. 

## Testing instructions
- Run `npm install` in this branch.
- Enable the ECE feature flag (return `true` from `is_stripe_ece_enabled` function or in your database add an option `_wcstripe_feature_ece` and set `yes` as the value)
- Go to the block checkout page and confirm that the ECE button is rendered.

<img width="885" alt="Screenshot 2024-09-05 at 1 42 29 AM" src="https://github.com/user-attachments/assets/161ae7f8-64f9-447f-9924-0b6e19d106d2">

- Disable the ECE feature flag.
- Confirm that the PRB button is rendered on both the block and shortcode checkout page. 

<img width="885" alt="Screenshot 2024-09-05 at 1 51 48 AM" src="https://github.com/user-attachments/assets/9e2f98df-a37f-47b2-a198-415a7f42efd2">
